components/movementsensor/replay/replay.go

```diff
@@ -0,0 +1,450 @@
+// Package replay implements a replay movement sensor that can return motion data.
+package replay
+
+import (
+	"context"
+	"sync"
+	"time"
+
+	"github.com/golang/geo/r3"
+	geo "github.com/kellydunn/golang-geo"
+	"github.com/pkg/errors"
+
+	"github.com/edaniels/golog"
+	datapb "go.viam.com/api/app/data/v1"
+	"go.viam.com/rdk/components/movementsensor"
+	"go.viam.com/rdk/internal/cloud"
+	"go.viam.com/rdk/resource"
+	"go.viam.com/rdk/spatialmath"
+	goutils "go.viam.com/utils"
+	"go.viam.com/utils/rpc"
+	"google.golang.org/protobuf/types/known/timestamppb"
+)
+
+const (
+	timeFormat            = time.RFC3339
+	grpcConnectionTimeout = 10 * time.Second
+	downloadTimeout       = 30 * time.Second
+	maxCacheSize          = 100
+)
+
+var (
+	// model is the model of a replay movement sensor.
+	model = resource.DefaultModelFamily.WithModel("replay")
+
+	// ErrEndOfDataset represents that the replay sensor has reached the end of the dataset.
+	ErrEndOfDataset = errors.New("reached end of dataset")
+
+	defaultCacheStatus = map[string]bool{
+		"position":         false,
+		"orientation":      false,
+		"angular_velocity": false,
+		"linear_velocity":  false,
+		"compass_heading":  false,
+		"orientation":      false,
+	}
+)
+
+func init() {
+	resource.RegisterComponent(movementsensor.API, model, resource.Registration[movementsensor.MovementSensor, *Config]{
+		Constructor: newReplayMovementSensor,
+	})
+}
+
+// Validate checks that the config attributes are valid for a replay movement sensor.
```

Comment:
[note] same as replay camera

Reply:
Note: robot_id, org_id and location_id are not required attributes currently. Should they be?

Reply:
Yes, they should be

Reply:
Done